engine/src/configuration/graphics_config.h

```diff
@@ -28,18 +29,66 @@
 #ifndef VEGA_STRIKE_ENGINE_PYTHON_CONFIG_GRAPHICS_CONFIG_H
 #define VEGA_STRIKE_ENGINE_PYTHON_CONFIG_GRAPHICS_CONFIG_H
 
+// Core JSON-based utility header
+#include "json.h"
```

Comment:
What directory is json.h located in? The CI builds are failing with error messages stating that this file cannot be found.

Other than that, this PR looks excellent. Outstanding work, @ollobrains , and thanks for the contribution!